src/doc-utils.js

```diff
@@ -1,34 +1,35 @@
 "use strict";
 
 function traverseDoc(doc, onEnter, onExit, shouldTraverseConditionalGroups) {
-  var hasStopped = false;
```

Comment:
hasStopped is a too aggressive for our new use case, we want to avoid going down recursively, but not stop the traversal altogether. Fortunately, we can change the behavior where `false` means that we should stop recursing and implement the hasStopped behavior in the call site that needs it.